src/posts/2020-04-16-continuous-stuff-with-github-actions/index.md

```diff
@@ -0,0 +1,268 @@
+GitHub to create a deployment, by providing a certain git reference, an environment to which you want to deploy, and
+other optional parameters.
+
+GitHub will then collect this and build a history of all the deploys you requested, and the status of each one (which
+can be updated using the same API).
+
+This API won't really do anything by itself, though. It builds a nice log, but that's about it.  You can subscribe to
+webhooks from this API though.
+
+So any 3rd party service could theoretically listen to these webhooks, and process the deployment you requested, instead
+of forcing you the come up with fancy ways to commit things in a particular way.
+
+None of these CIs seem to do that, though. And that's why GitHub Actions are so different.
+
+# GitHub Action triggers
+
+GitHub's new toy allows us to build jobs that can be triggered [in a variety of ways][github-workflow-triggers]. You can
+have the traditional "on every push" jobs, but also more fancy stuff, such as "on every deployment created", or even "on
+every commit to an issue".
+
+```yaml
+name: GitHub Action Example
+
+on: [deployment]
+
+jobs:
+  steps:
+    # ...
+    - run: npm run deploy ${{ github.event.deployment.environment }}
+    # ...
+```
+
+This sample does not rely on a `git push` in any way. All that is needed to trigger it is to call the Deployments API,
+and the rest of the job can pick up parameters from the webhook's data to figure out what to do, such as what
+environment to deploy to.
+
+## Preventing redundant builds
+
+One other thing that other CIs also aimed to achieve, is to optimize usage by preventing branches from triggering builds
+unless a Pull Request is open for them. The rationale here is that a work-in-progress branch rarely needs CI feedback,
+and would only waste resources. This is often a configurable option in CI settings. But with GitHub action triggers, we
+can do a bit more than that, by having that configuration in the source code itself:
+
+```yaml
+name: GitHub Action Example
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - '*'
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  # ...
+```
+
+We might want to always trigger builds for the master branch, or for every tag that is created. But for pull requests
+(which have their own webhook, and therefore, trigger), we can specify that only creation or updates to those Pull
+Requests' history should trigger the job.
+
+## Manual workflows with comments
+
+We can also react to comments on the Pull Request itself, which can be useful if you want a more seamless integration of
+some features.
+
+In my case, I wanted to deploy a preview version of Pull Requests to our frontend application.  It wasn't efficient to
+do this for every single PR though (only a small subset of them actually need this), so we went with this instead:
+
+![Comments triggering GitHub Action](./github-actions-comment.png)
```

Comment:
Wait, what causes the trigger? Is it the 🚀 emoji? Is it a simple comment. Or the "preview" keyword?

Reply:
uh. seems I accidentally deleted one sentence here. I distinctly remember writing it

Will add back. but FYI: it's the "preview" keyword. the emoji is the action answering back to acknowledge it got the message